WordPress/openverse-frontend
# Stop generating pot file on every test run

## Fixes


Fixes #320 by @sarayourfriend 

## Description


This PR fixes the bug when a pot file was generated on every push.
Pre-commit runs test before every push. The tests for pot file generation were importing from `src/locales/scripts/json-to-pot.js` file. Since the file also had the call to generate the .pot file on the top level, it was creating a new `openverse.pot` file on every import. The new file's only difference was the timestamp.
Here, the helper functions necessary to generate a `pot` file were moved out from the script that is run by `npm run i18n:generate-pot` and is generating the `pot` file.

## Testing Instructions

Create a new branch on top of this one, add a small change, commit it and push. When you run `git status` after that, you will see no other changes. Unlike the current main branch, which will show a modified `openverse.pot` file.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
